metaflow/plugins/kfp/kfp.py

```diff
@@ -998,23 +740,67 @@ def build_kfp_dag(
                     METAFLOW_USER=METAFLOW_USER,
                 )
                 metaflow_run_id = f"kfp-{dsl.RUN_ID_PLACEHOLDER}"
-                step_op_args = dict(
-                    cmd_template=kfp_component.cmd_template,
-                    metaflow_run_id=metaflow_run_id,
-                    metaflow_configs=metaflow_configs,
-                    passed_in_split_indexes=passed_in_split_indexes,
-                    preceding_component_inputs=preceding_component_inputs,
-                    preceding_component_outputs=kfp_component.preceding_component_outputs,
-                    flow_parameters_json=flow_parameters_json
-                    if node.name == "start"
-                    else None,
+
+                command = [
+                    "bash",
+                    "-ec",
+                    kfp_component.init_cmd
+                    + " && python -m metaflow.plugins.kfp.kfp_step_function"
+                    + f" --metaflow_run_id {metaflow_run_id}"
+                    + f" --metaflow_configs {json.dumps(json.dumps(metaflow_configs))}"
+                    + f' --cd_into_metaflow_package_cmd "{kfp_component.cd_into_metaflow_package_cmd}"'
+                    + f' --clean_volume_cmd "{kfp_component.clean_volume_cmd}"'
+                    + f" --task_id {kfp_component.task_id}"
+                    + f' --task_id_template "{kfp_component.task_id_template}"'
+                    + f" --step_name {kfp_component.step_name}"
+                    + f" --flow_name {kfp_component.flow_name}"
+                    + f" --tags {json.dumps(json.dumps(kfp_component.tags))}"
+                    + f" --environment_type {kfp_component.environment_type}"
+                    + f" --logger_type {kfp_component.logger_type}"
+                    + f" --monitor_type {kfp_component.monitor_type}"
+                    + f" --user_code_retries {kfp_component.user_code_retries}"
+                    + " --workflow_name {{workflow.name}}"
+                    + f" --script_name {os.path.basename(sys.argv[0])}"
+                    + f" --passed_in_split_indexes={passed_in_split_indexes}"
+                    + f" --preceding_component_inputs={json.dumps(json.dumps(preceding_component_inputs))}"
+                    + f" --preceding_component_outputs={json.dumps(json.dumps(kfp_component.preceding_component_outputs))}",
+                ]
+
+                if node.name == "start":
+                    command[-1] += f" --flow_parameters_json='{flow_parameters_json}'"
```

Comment:
it's cleaner to be explicit and use `command.append()`

Reply:
I'm actually appending to a `string`, not a list (`command[-1]` is a string). To my knowledge, there isn't an `append()` function for strings.

Reply:
In this case could you please move line 795 below line 820, so that your kfp cli command is fully formatted. Editing `command[-1]` feels a bit implicit in my opinion.

Alternatively does line 808 to line 819 fit into `_generate_metaflow_execution_cmd`?